systemtests/src/main/java/io/enmasse/systemtest/messagingclients/MessagingClientRunner.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.enmasse.systemtest.messagingclients;
+
+import io.enmasse.api.model.MessagingEndpoint;
+import io.enmasse.api.model.MessagingEndpointPort;
+import io.enmasse.systemtest.Endpoint;
+import io.enmasse.systemtest.logs.CustomLogger;
+import io.enmasse.systemtest.messagingclients.rhea.RheaClientReceiver;
+import io.enmasse.systemtest.messagingclients.rhea.RheaClientSender;
+import io.enmasse.systemtest.messaginginfra.resources.MessagingAddressResourceType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+import static io.enmasse.systemtest.messaginginfra.resources.MessagingAddressResourceType.getPort;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MessagingClientRunner {
+    Logger LOGGER;
+
+    public MessagingClientRunner() {
+        LOGGER = CustomLogger.getLogger();
+    }
+
+    public List<ExternalMessagingClient> sendAndReceive(MessagingEndpoint endpoint, boolean waitReceivers, String senderAddress, String ... receiverAddresses) throws Exception {
+        return sendAndReceive(endpoint, waitReceivers, null, null, senderAddress, receiverAddresses);
+    }
+
+    public List<ExternalMessagingClient> sendAndReceive(MessagingEndpoint endpoint, String senderAddress, String ... receiverAddresses) throws Exception {
+        return sendAndReceive(endpoint, false, null, null, senderAddress, receiverAddresses);
+    }
+
+    /**
+     * Send 10 messages on sender address, and receive 10 messages on each receiver address.
+     */
+    public List<ExternalMessagingClient> sendAndReceive(MessagingEndpoint endpoint, boolean waitReceivers, Map<ClientArgument, Object> extraSenderArgs,
+                           Map<ClientArgument, Object> extraReceiverArgs, String senderAddress, String ... receiverAddresses) throws Exception {
+        int expectedMsgCount = 10;
+
+        ExecutorService executor = Executors.newFixedThreadPool(1 + receiverAddresses.length);
```

Comment:
This should be shutdown and awaited after clients are finished (try - finally) so that resources are cleaned up.

Reply:
Good catch, I have forget about that!

Reply:
Should be fixed now :) Can you check it @lulf  ?